test/functional/bulk.test.js

```diff
@@ -1625,6 +1628,64 @@ describe('Bulk', function () {
       });
   });
 
+  it('properly accounts for bson size in bytes in bulk ordered inserts', function (done) {
+    const client = this.configuration.newClient();
+    const size = MAX_BSON_SIZE / 2;
+    const largeString = crypto.randomBytes(size - 100).toString('hex');
+    const documents = [{ s: largeString }, { s: largeString }];
+
+    let db;
+
+    client
+      .connect()
+      // NOTE: Hack to get around unrelated strange error in bulkWrites for right now.
+      .then(() => {
+        db = client.db(this.configuration.db);
+        return db.dropCollection('doesnt_matter').catch(() => {});
+      })
+      .then(() => {
+        return db.createCollection('doesnt_matter');
+      })
+      .then(() => {
+        const coll = db.collection('doesnt_matter');
+
+        coll.insertMany(documents, { ordered: true }, err => {
+          client.close(() => {
+            done(err);
+          });
+        });
+      });
+  });
+
+  it('properly accounts for bson size in bytes in bulk unordered inserts', function (done) {
+    const client = this.configuration.newClient();
+    const size = MAX_BSON_SIZE / 2;
+    const largeString = crypto.randomBytes(size - 100).toString('hex');
+    const documents = [{ s: largeString }, { s: largeString }];
+
+    let db;
+
+    client
+      .connect()
+      // NOTE: Hack to get around unrelated strange error in bulkWrites for right now.
+      .then(() => {
+        db = client.db(this.configuration.db);
+        return db.dropCollection('doesnt_matter').catch(() => {});
+      })
+      .then(() => {
+        return db.createCollection('doesnt_matter');
+      })
+      .then(() => {
+        const coll = db.collection('doesnt_matter');
+
+        coll.insertMany(documents, { ordered: false }, err => {
+          client.close(() => {
+            done(err);
+          });
+        });
+      });
```

Comment:
```suggestion
      .then(coll => {
        return coll.insertMany(documents, { ordered: false });
      })
      .finally(() => client.close());
```
Just a suggestion but you could maintain the use of promises all the way down here, seems more consistent.
This is the opposite of what @emadum you suggest since it will still implicitly pass the potential error along to mocha, thoughts?

Reply:
That's a good point and SGTM, just using promises all the way down is probably more idiomatic. In addition to above suggestion the tests would also need to return the promise chain.

Reply:
I like this suggestion. I've updated.